src/core/utils/internal/parseGraphQLRequest.ts

```diff
@@ -167,39 +167,44 @@ async function getGraphQLInput(request: Request): Promise<GraphQLInput | null> {
   }
 }
 
+const parsedGraphQLRequestCache = new WeakMap<Request, ParsedGraphQLRequest>()
```

Comment:
A silly question here: why not lift the caching to the `*Handler` individual classes and make it apply for HTTP requests too? The issue you describes is in no way exclusive to the GraphQL handler, it simply surfaces there better. 

I think this is a good proposal, and we should add something like `private cache: WeakMap<Request, X>` onto the `HttpRequest`/`GraphQL` request classes. I don't think this belongs to the generic `RequestHandler` though.

Reply:
Oh, a great point!

This really falls under a bit larger refactoring I had in mind for `getResponse`. Maybe we can discuss and kick off _that_? 

Regarding #2, the caching would happen in the `.run()` method anyway. This way the base class would ensure consistent caching across different overrides of the parse/predicate methods. 

I see that the cache should come from above. That's why I mentioned the refactoring. I think the `getResponse()` function should be made into a class. An orchestrator of sorts. It would hold the cache, perhaps.